packages/catalog-process/src/services/catalogService.ts

```diff
@@ -625,212 +548,211 @@ export function catalogServiceBuilder(
         })
       );
     },
-  };
-}
 
-export function createEserviceLogic({
-  eserviceWithSameName,
-  apiEServicesSeed,
-  authData,
-}: {
-  eserviceWithSameName: WithMetadata<EService> | undefined;
-  apiEServicesSeed: ApiEServiceSeed;
-  authData: AuthData;
-}): CreateEvent<EServiceEvent> {
-  if (authData.externalId.origin !== "IPA") {
-    throw originNotCompliant("IPA");
-  }
+    async createEService(
+      apiEServicesSeed: ApiEServiceSeed,
+      authData: AuthData
+    ): Promise<EService> {
+      logger.info(
+        `Creating EService with service name ${apiEServicesSeed.name}`
+      );
 
-  if (eserviceWithSameName) {
-    throw eServiceDuplicate(apiEServicesSeed.name);
-  }
+      if (authData.externalId.origin !== "IPA") {
+        throw originNotCompliant("IPA");
+      }
 
-  const newEService: EService = {
-    id: generateId(),
-    producerId: authData.organizationId,
-    name: apiEServicesSeed.name,
-    description: apiEServicesSeed.description,
-    technology: apiTechnologyToTechnology(apiEServicesSeed.technology),
-    mode: apiEServiceModeToEServiceMode(apiEServicesSeed.mode),
-    attributes: undefined,
-    descriptors: [],
-    createdAt: new Date(),
-    riskAnalysis: [],
-  };
+      const eserviceWithSameName =
+        await readModelService.getEServiceByNameAndProducerId({
+          name: apiEServicesSeed.name,
+          producerId: authData.organizationId,
+        });
```

Comment:
We were passing this as a parameter. Now we could retrieve it at the moment it's needed.